### PR TITLE
Crud actions

### PR DESCRIPTION
**Pull request by nikkinikayly**

### CODE REVIEW

File: app/controllers/api/categories_controller.rb [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
   end
 
   def show
+    render json: @category
+  end
+
+  def create
+    
+    category = @menu.categories.new(category_params)
+    # binding.pry
```
Comment (kdeep97) [inline app/controllers/api/categories_controller.rb]: remove binding.pry

File: app/controllers/api/categories_controller.rb [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
   def update
+    if @category.update(category_params)
+      render json: @category 
+    else
+      render json: @category.errors, status: 422
+    end
   end
 
   def destroy
+    binding.pry
```
Comment (kdeep97) [inline app/controllers/api/categories_controller.rb]: remove binding.pry

File: app/controllers/api/items_controller.rb [hunk 3]
```diff
@@ -1,16 +1,52 @@
...
   end
 
   def show
+    render json: @item
+  end
+
+  def create
+    
+    item = @categories.items.new(item_params)
+    # binding.pry
```
Comment (kdeep97) [inline app/controllers/api/items_controller.rb]: remove

File: app/controllers/api/items_controller.rb [hunk 4]
```diff
@@ -1,16 +1,52 @@
...
   def update
+    if @item.update(item_params)
+      render json: @item 
+    else
+      render json: @item.errors, status: 422
+    end
   end
 
   def destroy
+    binding.pry
```
Comment (kdeep97) [inline app/controllers/api/items_controller.rb]: remove

File: client/src/components/Categories.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    axios.post(`/api/menus/${this.props.menuId}/categories`, {category})
+      .then(res => {
+        this.setState({categories: [...this.state.categories, res.data]})
+      })
+  }
+
+  deleteCat = (id) => {
+    let menId = parseInt(this.props.menuId)
+    console.log(menId)
+    // debugger
```
Comment (kdeep97) [inline client/src/components/Categories.js]: remove debugger

File: client/src/components/Categories.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      axios.delete(`/api/menus/${menId}/categories/${id}`)
+          .then(res => {
+            debugger
```
Comment (kdeep97) [inline client/src/components/Categories.js]: remove

File: client/src/components/Categories.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        }) 
+        .catch(res => {
+          debugger
```
Comment (kdeep97) [inline client/src/components/Categories.js]: remove

File: client/src/components/EditMenu.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  state = { menu: {}, categories: [], editName: false, addCat: false }
+
+  componentDidMount() {
+    axios.get(`/api/menus/${this.props.match.params.id}`)
+      .then( res => {
+        this.setState({ menu:res.data })
+      })
+  }
+
+  deleteMenu = (id, e) => {
```
Comment (kdeep97) [inline client/src/components/EditMenu.js]: the delete function does not need e, only the id. Form submission functions are the ones that need e

**Comment by ColtenVanTussenbrook** [comment]
Oh I haven't gone through this...I probably shouldn't have pushed it to my branch yet. I wouldn't merge this to Master yet

**Pull request closed without merging**